power: stm32wb: support of the Low Power Mode

FRASTM opened the pull request:
This patch enables the low power Mode for the stm32wb soc. 
This is configured through the SYS_POWER_MANAGEMENT.
It uses the LPTIM timer as tickless timer (STM32_LPTIM_TIMER) to count the sleep period and wakeup the kernel on lptim interrupt.
The SLEEP 1,2,3 modes are supported on this device, which correspond to the stm32 STOP 0,1,2 modes.
Basically, there is a configuration maximum timeout allowed by the lptim is 2000ms when clocked by the LSE. 

The Deep Sleep mode is not supported by this PR.

Signed-off-by: Francois Ramu francois.ramu@st.com

FRASTM commented:
rebase and adapt on top of  https://github.com/zephyrproject-rtos/zephyr/pull/19026

FRASTM commented:
remove blank line

erwango reviewed (changes requested):
Please fix commit message ("# Conflicts: ....")

FRASTM committed:
soc: arm: st_stm32 add low power to stm32wb series

This patch introduces the support of low power modes
for the STM32WBxx from STMicroelectronics based on the lptim
Here, the power modes are sleep modes have lptimer as wakeup source.
The sleep modes are configured by the SYS_POWER_MANAGEMENT.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
stm32wb low power modes depends on the CONFIG_SYS_POWER_MANAGEMENT

FRASTM commented:
move the

config STM32_LPTIM_TIMER
	bool
	default y
	depends on SYS_POWER_MANAGEMENT

from soc/arm/st_stm32/common/Kconfig.defconfig.series to soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series

erwango reviewed (changes requested):
One last comment.

FRASTM commented on subsys/power/device.c:
```diff
@@ -45,8 +45,8 @@ static const char *const core_devices[] = {
-#elif defined(CONFIG_SOC_SERIES_STM32L4X)
-#define MAX_PM_DEVICES	1
+#elif defined(CONFIG_SOC_SERIES_STM32L4X) || defined(CONFIG_SOC_SERIES_STM32WBX)
+#define MAX_PM_DEVICES	2
```
setting 2 PM_DEVICES for WB and L4 soc, except if MAX_PM_DEVICES means something else

erwango replied:
I don't actually understand why this is increased when adding WB support.
If this is 1 for L4, it should be 1 for WB, right ?

FRASTM replied:
changing 1 for L4 or WB

ABOSTM reviewed (commented):
File power.c for WB looks very similar to L4 file. Is there no way to have a common file ?

erwango commented:
Sanitychek failure: ' Sanitycheck / qemu_arc_hs:tests/portability/cmsis_rtos_v1/portability.cmsis_rtos_v1 / tests/portability/cmsis_rtos_v1/portability.cmsis_rtos_v1'

FRASTM commented:
rebase
